[Teh Xue Yong] iP

xuey0ng opened the pull request.

Commit:
Add Gradle support

xuey0ng committed:
Added Greet, Echo, Exit in Duke.java

xuey0ng committed:
Added Add, List functionality to Duke.java

xuey0ng committed:
Updated .gitignore

xuey0ng committed:
Added task.java to track tasks and the ability to mark tasks as done.

xuey0ng committed:
Refactored Code and Added ToDos, Events, Deadlines

xuey0ng committed:
Modified runtest.bat and EXPECTED.TXT for Automated Text UI Testing

xuey0ng committed:
Added Error Handling

xuey0ng committed:
Added Delete functionality to Duke.java

xuey0ng committed:
Refactored code to use enums

xuey0ng committed:
Refactored Duke.java to use the new enums class

xuey0ng committed:
Add Save functionality to bot

Tasks are not persistent between runs.

The user is unable to view tasks after reloading the bot.

Update Task, Deadline, Event, Todo to support Save
Create Storage class to read, write tasks
Change Duke to allow for save
Create CorruptedFileException for corrupted files.

Allow for abstraction and inheritance.

xuey0ng committed:
Add dates

Modify Event and Deadline to store dates
Modify Duke to display user guide
Modify Duke to check for valid date

xuey0ng committed:
Merge branch 'branch-Level-7'

xuey0ng committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

xuey0ng committed:
Fix bug from merge with branch-Level-8

Modify Storage to fix invalid type bug

xuey0ng committed:
Refactor code to incorporate more OOP

Duke class is handling more than one responsibility.

Extract Ui, Storage, Parser, TaskList
Extract Command
Add ListCommand, ExitCommand, DoneCommand, TodoCommand, EventCommand,
DeadlineCommand, DeleteCommand.

Extracting the common behavior into an abstract class allows us to use
polymorphism while abstracting common information.

xuey0ng committed:
Refactor classes into packages

xuey0ng committed:
Add JUnit tests

xuey0ng committed:
Add JavaDoc to files

xuey0ng committed:
Add Find functionality to Dude

Add FindCommand
Modify Parser to accomodate new command
Modify Task to return description.

xuey0ng committed:
Modify Command files to match FindCommand

xuey0ng committed:
Merge branch 'branch-Level-9'

xuey0ng committed:
Add JavaDocs for FindCommand

xuey0ng committed:
Merge branch 'branch-A-JavaDoc'

xuey0ng committed:
Modify files to follow style guide

xuey0ng committed:
Merge branch 'branch-CodingStandard'

xuey0ng committed:
Modify text-ui-test to accomodate new main

xuey0ng committed:
Create local add-gradle-support branch

xuey0ng committed:
Merge branch 'add-gradle-support'

xuey0ng committed:
Move Main class to dude package

xuey0ng committed:
Update build.gradle to support Shadow

xuey0ng committed:
Add initial support for javaFX

xuey0ng committed:
Add CheckStyle

xuey0ng committed:
Add JavaDocs for all classes and methods

Zhang-Yunjie commented on src/main/java/Duke.java:
```diff
@@ -1,185 +1,46 @@
+
+    public void run() {
+        ui.greet();
+        boolean isExit = false;
```
Should "isExit" be "hasExit" instead? Perhaps it will sound more natural and grammatically correct if you intend for it to represent the state of Duke :)

xuey0ng replied:
I use isExit as the program has not exited and hence, the variable represents "is exiting: instead of "has exited". Both sound pretty rough if were to look at it from a grammatical point of view :)

Zhang-Yunjie commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,19 @@
+
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws CommandException;
+
+    public boolean isExit() {
```
Should it be named "getExitStatus()" instead of "isExit()" ? The action of retrieving exit status may be more explicit this way :)

xuey0ng replied:
Good point!

Zhang-Yunjie commented on src/main/java/EventCommand.java:
```diff
@@ -0,0 +1,31 @@
+
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws CommandException {
+        try {
+            StringBuilder str = new StringBuilder();
```
Could you think of more specific and descriptive names for your variables? "str", "temp" and "current" may be vague at the first glance :)

xuey0ng replied:
I'm thinking about whether to keep these scratch variables short as they are not returned, and are not used outside 2 -3 lines of code.

Zhang-Yunjie commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,34 @@
+            Commands current;
+            try {
+                current = Commands.valueOf(commands[0].toUpperCase());
+                switch(current) {
```
Correct indentation for switch method, this is a common mistake. Good job 👍  !

xuey0ng replied:
thanks :)

Zhang-Yunjie commented on src/main/java/dude/command/Command.java:
```diff
@@ -5,6 +5,10 @@
+/**
+ * The abstract task contains the action of the command and whether the command calls for an exit.
+ */
+
```
Should you remove this empty line between class header comment and the class itself?

Zhang-Yunjie commented:
LGTM, almost. I like how your OO structure is designed to be clear and easy to understand :)

xuey0ng committed:
Change variables name from PR review input

xuey0ng committed:
Add partial implementation of javaFX

ChenXJ98 commented on src/main/java/dude/command/Command.java:
```diff
@@ -0,0 +1,30 @@
+
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws CommandException;
+
+    public boolean getExitStatus() {
```
Maybe can just name this method as isExit or getIsExit. Personally I feel its more clear that way 👍

ChenXJ98 reviewed (commented):
Looks really good overall! Just a nit pick :)

xuey0ng committed:
Add full javaFx user interface functionality

xuey0ng committed:
Fix CheckStyle warnings

xuey0ng committed:
Cleanup code

xuey0ng committed:
Fix hidden overflow in GUI

xuey0ng committed:
Merge branch 'branch-Level-10'

xuey0ng committed:
Add assertions

xuey0ng committed:
Change variable names

xuey0ng committed:
Merge branch 'branch-A-Assertions'

xuey0ng committed:
Add functionality to detect duplicate entries

xuey0ng committed:
Merge branch 'C-DetectDuplicates'

xuey0ng committed:
Improve GUI

Add background gradient
Add different visual format
Add color for error messages
Improve send button
Improve textfield

xuey0ng committed:
Update README

xuey0ng committed:
Change find command to be case-insensitive

xuey0ng committed:
Change find command to ignore order of keywords

xuey0ng committed:
Update User Guide in docs\README.md

xuey0ng committed:
Add Ui.png to docs

xuey0ng committed:
Add support for alternate ui

xuey0ng committed:
Merge pull request #1 from fall9x/test-features

Add support for alternate ui

xuey0ng committed:
Add javadocs for the newer methods

xuey0ng committed:
Fix checkstyle erros in test

xuey0ng committed:
Add Ui.png.PNG

xuey0ng committed:
Change Ui.PNG to Ui.png

xuey0ng committed:
Change .png to .PNG extension

xuey0ng committed:
Change '\' to '/' for image paths

xuey0ng committed:
Delete Ui.PNG

xuey0ng committed:
Fix Ui.png
